src/Form/Control/TreeItemSelector.php

```diff
@@ -112,6 +120,36 @@ public function setTreeItems(array $treeItems)
         return $this;
     }
 
+    /**
+     * @param mixed $parentId
+     */
+    protected function addNodes(Model $model, $parentId = null): array
+    {
+        $result = [];
+
+        $nodeModel = (clone $model)->addCondition($this->parentIdField, $parentId);
+
+        foreach ($nodeModel as $node) {
+            if ($node->get($this->parentIdField) === $parentId) {
+                $newNode = [];
+                $newNode['name'] = $node->getTitle();
+                $newNode['id'] = $node->getId();
+                $newNode['parent_id'] = $node->get($this->parentIdField);
+                $newNode['nodes'] = $this->addNodes(clone $model, $node->getId());
```

Comment:
why to clone here?

Reply:
Because we have to set a new parentId condition on an unconditioned model, otherwise will have contradictory conditions

Reply:
Correct myself - this clone is indeed not needed, as we clone earlier into `nodeModel` already. Correct in latest commit